docs/data-structures.rst

```diff
@@ -128,54 +114,38 @@ Gavel Validator Output (object)
+   -  expected (any) - Expected value of an HTTP message field
+   -  actual (any) - Actual value of an HTTP message field
 
-The validation error is based on format provided by `Amanda <https://github.com/apiaryio/Amanda>`__ and is also “documented” `here <https://github.com/apiaryio/Amanda/blob/master/docs/json/objects/error.md>`__. Although for validation of draft4 JSON Schema Gavel uses `tv4 <https://github.com/geraintluff/tv4>`__ library, the output then gets reshaped into the structure of Amanda’s errors.
+- errors (:ref:`gavel-error`)
 
-This validation result is returned not only when validating against `JSON Schema`_, but also when validating against JSON example or when validating HTTP headers.
+.. _gavel-error:
 
--  length: ``0`` (number, default) - number of error properties
--  errorMessages (object) - doesn’t seem to ever contain anything or be used for anything
--  *0* (object) - validation error details, property is always a string containing a number (0, 1, 2, …)
+Gavel Error (object)
+--------------------
 
-   -  property (array[string]) - path to the problematic property in format of `json-pointer’s parse() output <https://github.com/manuelstofer/json-pointer#user-content-parsestr>`__
-   -  propertyValue (mixed) - real value of the problematic property (can be also ``undefined`` etc.)
-   -  attributeName: ``enum``, ``required`` (string) - name of the relevant JSON Schema attribute, which triggered the error
-   -  attributeValue (mixed) - value of the relevant JSON Schema attribute, which triggered the error
-   -  message (string) - error message (in case of tv4 it contains :rfc:`JSON Pointer <6901>` to the problematic property and for both Amanda and tv4 it can directly mention property names and/or values)
-   -  validator: ``enum`` (string) - the same as ``attributeName``
-   -  validatorName: ``error``, ``enum`` (string) - the same as ``attributeName``
-   -  validatorValue (mixed) - the same as ``attributeValue``
+Can also be seen as part of Gavel Validator Output `here <https://relishapp.com/apiary/gavel/docs/data-validators-and-output-format#validators-output-format>`__.
 
-.. _textdiff-validation-result:
+-  message (string) - Error message
+-  location (object) - (Optional) kind-dependent extra error information
 
-TextDiff Validation Result (string)
------------------------------------
+   -  pointer (string) - :rfc:`JSON Pointer <6901>` path
+   -  property (string[]) - A deep property path
 
-Block of text which looks extremely similar to the standard GNU diff/patch format. Result of the ``patch_toText()`` function of the ``google-diff-match-patch`` library (`docs <https://github.com/google/diff-match-patch/wiki/API#user-content-patch_totextpatches--text>`__).
+.. _test-run-error:
 
-.. _gavel-error:
+Test run error (object)
+-----------------------
 
-Gavel Error (object)
---------------------
+Whenever an exception occurs during a test run it's being recorded under the ``errors`` property of the test.
```

Comment:
It's not just exceptions, I believe there are various errors and warnings. That's why these do preserve the `severity` property (undocumented here).

The error is set e.g. when user sets `transaction.skip = 'Not implemented yet'` in hooks or something similar. That's going to get propagated as

```js
errors = [{message: 'Not implemented yet', severity: 'warning'}]
```

This section needs more love.